kbase-extension/static/kbase/js/widgets/narrative_core/upload/stagingAreaViewer.js

```diff
@@ -309,76 +334,96 @@ define([
          */
         renderFiles: function (files) {
```

Comment:
This is where most of the heaviest changes here.
Some highlights:
* file data gets fed into the table directly, not from the DOM
* file paths are used throughout, not just names
* the rowCallback function was a bit of a nightmare of nested function definitions. I decomposed that out into widget-level functions for clarity.
* now makes use of `createdCell` and `createdRow` to make DOM changes (like classes and attributes)